### PR TITLE
Optimize detecting relevant changed files

### PR DESCRIPTION
**Pull request by mathjazz**
Detecting which on the list of changed files since the last sync is
an actual resource file and which locale does it belong to could take
minutes, now it should take seconds.

@jotes r?

This is a quite significant improvement. Due to this bottleneck, sync sometimes runs significantly faster with the `--force` flag (which syncs all resources)!

### CODE REVIEW

File: pontoon/sync/vcs/models.py [hunk 1]
```diff
@@ -151,36 +151,22 @@ def find_changed_files(repo, locale=None):
...
                 for changed_file in find_changed_files(repo):
-                    path_info = self.get_path_info(changed_file, repo)
+                    if is_hidden(changed_file):
+                        continue
 
-                    if path_info:
-                        path, locale_path, locale = path_info
-                        path = path[len(locale_path):].lstrip(os.sep)
-                        files.setdefault(path, []).append(locale)
+                    for ldp in locale_directories_paths:
```
Comment (jotes) [inline pontoon/sync/vcs/models.py]: Erm, right... Good idea.  `get_path_info` is a legacy from the one of the more "functional" versions of this code.9

File: pontoon/sync/vcs/models.py [hunk 2]
```diff
@@ -151,36 +151,22 @@ def find_changed_files(repo, locale=None):
                     for path in find_changed_files(repo, locale):
                         files.setdefault(path, []).append(locale)
             else:
+                locale_directories = self.locale_directories(repo)
+                locale_directories_paths = locale_directories.keys()
+
                 for changed_file in find_changed_files(repo):
-                    path_info = self.get_path_info(changed_file, repo)
+                    if is_hidden(changed_file):
```
Comment (jotes) [inline pontoon/sync/vcs/models.py]: Hm, do you have any exaples of repos that have a large number of hidden files in their structure ?

**Comment by jotes** [comment]
A small step for code, a giant leap for pontoon. r+ ;D

File: pontoon/sync/vcs/models.py (hunk 2 above)
Comment (mathjazz) [inline pontoon/sync/vcs/models.py]: I don't think so. Are you implying we should remove this check?

**Pull request merged**